src/files/ThemeXml.ts

```diff
@@ -82,68 +114,49 @@ export class ThemeXml extends XmlFile {
-		location = location ?? 'word/theme/theme1.xml';
-		const themeDocument = archive.hasFile(location) ? await archive.readXml(location) : xml;
-		const fontScheme: Record<string, string | Font[]> = evaluateXPathToMap(`
-			//${QNS.a}theme/${QNS.a}themeElements/${QNS.a}fontScheme/map {
-				"name": @name/string(),
-				"majorFontLatinTypeface": ${QNS.a}majorFont/${QNS.a}latin/@typeface/string(),
-				"majorFontLatinPanose": ${QNS.a}majorFont/${QNS.a}latin/@panose/string(),
-				"majorFontOthers": array{${QNS.a}majorFont/${QNS.a}font/map { "script": @script/string(), "typeface": @typeface/string()}},
-				"minorFontLatinTypeface": ${QNS.a}minorFont/${QNS.a}latin/@typeface/string(),
-				"minorFontLatinPanose": ${QNS.a}minorFont/${QNS.a}latin/@panose/string(),
-				"minorFontOthers": array{${QNS.a}minorFont/${QNS.a}font/map { "script": @script/string(), "typeface": @typeface/string()}}
-			}
-		`, themeDocument);
-
-		const majorFontOthersCollection: Font[] = typeof fontScheme["majorFontOthers"] !== 'string'
-			? fontScheme["majorFontOthers"].map((font: Font) => {
-				return { script: font["script"], typeface: font["typeface"] } as Font
-			}) : [{ script: '', typeface: '' } as Font];
-
-
-		const minorFontOthersCollection: Font[] = typeof fontScheme["minorFontOthers"] !== 'string'
-			? fontScheme["minorFontOthers"].map((font: Font) => {
-				return { script: font["script"], typeface: font["typeface"] } as Font
-			}) : [{ script: '', typeface: '' } as Font];
-
-		const newFontScheme: FontScheme = {
-			name: fontScheme["name"] as string,
-			majorFont: {
-				latinFont: {
-					typeface: fontScheme["majorFontLatinTypeface"],
-					panose: fontScheme["majorFontLatinPanose"]
-				} as LatinFont,
-				otherFonts: majorFontOthersCollection
+	public static fromDom(dom: Document, location: string): Promise<ThemeXml> {
+		const fontScheme = evaluateXPathToMap<FontScheme>(`
+		./${QNS.a}theme/${QNS.a}themeElements/${QNS.a}fontScheme/map {
+			"majorFont": map {
+				"latinFont": map {
+					"typeface": ${QNS.a}majorFont/${QNS.a}latin/@typeface/string(),
+					"panose":  ${QNS.a}majorFont/${QNS.a}latin/@panose/string()
+				},
+				"otherFonts": array{${QNS.a}majorFont/${QNS.a}font/map { "script": @script/string(), "typeface": @typeface/string()}}
 			},
-			minorFont: {
-				latinFont: {
-					typeface: fontScheme["minorFontLatinTypeface"],
-					panose: fontScheme["minorFontLatinPanose"]
-				} as LatinFont,
-				otherFonts: minorFontOthersCollection
+			"minorFont": map {
+				"latinFont": map {
+					"typeface": ${QNS.a}minorFont/${QNS.a}latin/@typeface/string(),
+					"panose": ${QNS.a}minorFont/${QNS.a}latin/@panose/string()
+				},
+				"otherFonts": array{${QNS.a}minorFont/${QNS.a}font/map { "script": @script/string(), "typeface": @typeface/string()}}
 			}
-		};
-
+		}`,
+		dom,
+		);
 		const newTheme = new ThemeXml(location);
-		newTheme.themeElements.fontScheme = newFontScheme;
+		newTheme.fontScheme = fontScheme;
```

Comment:
Ideally you'd use the setter for this, and benefit off of any data validation/grooming etc. that's already in there (or that would be there in the future)

But there could be a reason why you didn't, that I missed?